File: Export/Natives/Name/IFileNamePartProvider.cs
using FileNaming.CustomFileNaming;
using kCura.WinEDDS.Exporters;

namespace kCura.WinEDDS.Core.Export.Natives.Name
{
	public interface IFileNamePartProvider
	{
		string GetPartName(DescriptorPart descriptorDescriptorPartBase, ObjectExportInfo exportObject);
	}

	public interface IFileNamePartProvider<in T> : IFileNamePartProvider
	{
		string GetPartName(T descriptorPart, ObjectExportInfo exportObject);
	}
}

File: Source/Relativity.Import.Export.Legacy/FileNaming/CustomFileNaming/IFileNamePartProvider.vb
Imports FileNaming.CustomFileNaming
Imports kCura.WinEDDS.Exporters

Namespace kCura.WinEDDS.FileNaming.CustomFileNaming
	Public Interface IFileNamePartProvider

		Function GetPartName(descriptorDescriptorPartBase As DescriptorPart, exportObject As ObjectExportInfo) As String

	End Interface

	Public Interface IFileNamePartProvider(Of In T)
		Inherits IFileNamePartProvider

		Function GetPartName(descriptorPart As T, exportObject As ObjectExportInfo) As String

	End Interface
End Namespace
